plugins/modules/k8s_cluster_info.py

```diff
@@ -201,12 +202,12 @@ def execute_module(self):
         invalidate_cache = boolean(self.module.params.get('invalidate_cache', True), strict=False)
         if invalidate_cache:
             self.client.resources.invalidate_cache()
-        results = {}
+        results = defaultdict(list)
         for resource in list(self.client.resources):
             resource = resource[0]
             if isinstance(resource, ResourceList):
                 continue
-            results[resource.group] = {
+            results[resource.group].append({
```

Comment:
I feel like this data structure may be difficult to use from ansible as you'd have to search the list of resources. It may make more sense to key on `group -> version -> kind` which (except for openshift templates) should be unique. Ie, the structure could look like:

```
{
  "rbac.authorization.k8s.io": {
    "v1beta1": {
       "ClusterRole": {...},
       "ClusterRoleBinding": {...}
    },
    "v1": {
      "ClusterRole": {...},
      "ClusterRoleBinding": {...}
    }
  }
}
```

Or we could combine group and apiVersion (which is commonly done referencing resources), like:

{
  "rbac.authorization.k8s.io/v1": {
     "ClusterRole": {...},
     "ClusterRoleBinding": {...}
  }
  "rbac.authorization.k8s.io/v1beta1": {
     "ClusterRole": {...},
     "ClusterRoleBinding": {...}
  }
}
```

That way if I know the group + version + kind I can easily find the resource I want by key, rather than having to process a list of resources by subkey

Reply:
Thanks @fabianvf 
I will implement the second proposal.